src/Symfony/Component/HttpFoundation/Session/Storage/Handler/RedisSessionHandler.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Session\Storage\Handler;
+
+use Predis\Connection\Aggregate\ClusterInterface;
+use Predis\Response\Status;
+
+/**
+ * Redis based session storage handler based on the Redis class
+ * provided by the PHP redis extension.
+ *
+ * @author Dalibor Karlović <dalibor@flexolabs.io>
+ */
+class RedisSessionHandler extends AbstractSessionHandler
+{
+    private $redis;
+
+    /**
+     * @var int Time to live in seconds
+     */
+    private $ttl;
+
+    /**
+     * @var string Key prefix for shared environments
+     */
+    private $prefix;
+
+    /**
+     * List of available options:
+     *  * prefix: The prefix to use for the keys in order to avoid collision
+     *  * expiretime: The time to live in seconds. Defaults to 1 day.
+     *
+     * @param @param \Redis|\RedisArray|\RedisCluster|\Predis\Client $redisClient
+     * @param null|array                                             $options     An associative array of options
+     *
+     * @throws \InvalidArgumentException When unsupported client or options are passed
+     */
+    public function __construct($redis, ?array $options = null)
+    {
+        if (!$redis instanceof \Redis && !$redis instanceof \RedisArray && !$redis instanceof \Predis\Client && !$redis instanceof RedisProxy) {
+            throw new \InvalidArgumentException(\sprintf('%s() expects parameter 1 to be Redis, RedisArray, RedisCluster or Predis\Client, %s given', __METHOD__, \is_object($redis) ? \get_class($redis) : \gettype($redis)));
+        }
+
+        $this->redis = $redis;
+        $options = (array) $options;
```

Comment:
This was actually mentioned in https://github.com/symfony/symfony/pull/24781#discussion_r149052015, but since you're the second person to point it out, I've just changed it. :+1: